### PR TITLE
va-memorable-date: make month default to input instead of select

### PR DESCRIPTION
**Pull request by it-harrison**
## Chromatic

https://2316-three-inputs--60f9b557105290003b387cd5.chromatic.com

## Description
This PR changes the month default from a select to an input.

Closes [2316](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2316)

## QA Checklist
- [ ] Component maintains 1:1 parity with design mocks
- [ ] Text is consistent with what's been provided in the mocks
- [ ] Component behaves as expected across breakpoints
- [ ] Accessibility expert has signed off on code changes (if applicable. If not applicable provide reason why)
- [ ] Designer has signed off on changes (if applicable. If not applicable provide reason why)
- [ ] Tab order and focus state work as expected
- [ ] Changes have been tested against screen readers (if applicable. If not applicable provide reason why)
- [ ] New components are covered by e2e tests; updates to existing components are covered by existing test suite
- [ ] Changes have been tested in vets-website using Verdaccio (if applicable. If not applicable provide reason why)

## Screenshots

Screenshot 2024-01-10 at 5 21 13 PM

## Acceptance criteria
- [ ] QA checklist has been completed
- [ ] Screenshots have been attached that cover desktop and mobile screens

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Review by rmessina1010 — APPROVED** [review]
LGTM

File: packages/web-components/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@department-of-veterans-affairs/web-components",
-  "version": "4.53.0",
+  "version": "4.54.0",
```
Comment (jamigibbs) [inline packages/web-components/package.json:3]: The original PR had this as a `patch`.

File: packages/web-components/src/components/va-memorable-date/test/va-memorable-date.e2e.ts
Comment (jamigibbs) [inline packages/web-components/src/components/va-memorable-date/test/va-memorable-date.e2e.ts:1]: The [original PR](https://github.com/department-of-veterans-affairs/component-library/pull/869/files#diff-5f99ba08ddf6e7713450eeb5cece7ff70c2b67c773ad3a769f8a5a829d4fb4a0) had 406 changes (200 additions & 206 deletions) and this revert has 100 changes (13 additions & 87 deletions). I'm just double checking that we're not missing any tests.

File: packages/web-components/package.json (hunk 1 above)
Comment (it-harrison) [reply packages/web-components/package.json:3]: yeah, I wasn't sure which, I can change it to a patch

File: packages/web-components/src/components/va-memorable-date/test/va-memorable-date.e2e.ts
Comment (it-harrison) [reply packages/web-components/src/components/va-memorable-date/test/va-memorable-date.e2e.ts:1]: @jamigibbs I checked and all the tests are still there:
in `main`:
Screenshot 2024-01-11 at 8 45 06 PM
in this PR:
Screenshot 2024-01-11 at 8 40 13 PM

**Review by jamigibbs — APPROVED** [review]

**Pull request merged**